src/Language/Printer.php

```diff
@@ -226,6 +230,8 @@ public function printAST($ast)
                     },
 
                     NodeKind::NAMED_TYPE => static function (NamedTypeNode $node) : string {
+                        assert(is_string($node->name));
```

Comment:
I don't think we need this assertion (otherwise we may need similar assertion in many other methods as well)

Reply:
It satisfies Phpstan and is preferable to ignoring an error: If we change things around, it will trigger.

There is no runtime cost.

Reply:
> otherwise we may need similar assertion in many other methods as well

I know what you mean. This conversion seems to be the only one affected, since there is no string concatination (and thus automatic type coercion) involved.

It seems imperfect to me, but pragmatically it looks like our best option for now. https://github.com/webonyx/graphql-php/pull/697 would be a clean fix, until then this PR improves things a bit at least.

Reply:
What happens if we change:

```php
NodeKind::NAMED_TYPE => static function (NamedTypeNode $node) : string
```
to:
```php
NodeKind::NAMED_TYPE => static function ($modifiedNode) : string
```

As this is a more correct typing.

Reply:
That works fine, PHPStan no longer complains since it has nothing to work with.

Reply:
I think it is the most reasonable thing to do here as we deal with temporary objects, not full AST nodes. So please, go ahead and remove those type hints.